examples/otel-collector fails to build with ocb due to collector version drift

### Component

OBI (eBPF instrumentation)

### What happened?

The `examples/otel-collector` example failed to build with `ocb` due to version drift between the installed `ocb` binary, the hardcoded versions in `examples/otel-collector/builder-config.yaml`, and transitive Collector dependencies pulled through `go.opentelemetry.io/obi`.

Build logs -

  ```shell
  Error: failed to compile the OpenTelemetry Collector distribution: go subcommand failed with args '[build -trimpath -o otelcol-dev -ldflags=-s -w -gcflags=]': exit status 1, error message: #
  go.opentelemetry.io/collector/exporter/exporterhelper/xexporterhelper
  .../xexporterhelper@v0.151.0/profiles.go:66:11: undefined: queue.PersistRequestContextOnRead
  .../xexporterhelper@v0.151.0/profiles.go:84:11: undefined: queue.PersistRequestContextOnWrite
```

  This was fixed locally by aligning the versions in examples/otel-collector/builder-config.yaml to the same Collector release line as ocb v0.153.0.

### What did you expect to happen?

examples/otel-collector should build successfully by following the README steps.

### Steps to reproduce

1. Go to examples/otel-collector
2. Run:
```
./ocb --config ./builder-config.yaml
```
3. Observe the build failure above

### OBI configuration

```yaml
N/A. The failure happens during example build, before runtime.
```

### Relevant logs
```shell
2026-06-07T16:01:57.580+0530    INFO    internal/command.go:97  OpenTelemetry Collector Builder {"version": "v0.153.0+dirty"}
2026-06-07T16:01:57.581+0530    INFO    internal/command.go:102 Using config file       {"path": "./builder-config.yaml"}
2026-06-07T16:01:57.582+0530    INFO    builder/config.go:165   Using go        {"go-executable": "/opt/homebrew/bin/go"}
2026-06-07T16:01:57.584+0530    INFO    builder/main.go:107     Sources created {"path": "./otelcol-dev"}
2026-06-07T16:02:03.631+0530    INFO    builder/main.go:226     Getting go modules
2026-06-07T16:02:13.383+0530    INFO    builder/main.go:118     Compiling
Error: failed to compile the OpenTelemetry Collector distribution: go subcommand failed with args '[build -trimpath -o otelcol-dev -ldflags=-s -w -gcflags=]': exit status 1, error message: #
go.opentelemetry.io/collector/exporter/exporterhelper/xexporterhelper
.../xexporterhelper@v0.151.0/profiles.go:66:11: undefined: queue.PersistRequestContextOnRead
.../xexporterhelper@v0.151.0/profiles.go:84:11: undefined: queue.PersistRequestContextOnWrite
```

### Additional context

This appears to be an example maintenance issue rather than only a local setup issue.
There was also a Go toolchain wrinkle in my environment, but the core problem was that the example’s pinned Collector versions had drifted out of sync with the ocb release line.

Possible solutions:
- Documenting the expected ocb version in the README
- Adding CI coverage for `examples/otel-collector` so it can run `ocb --config ./builder-config.yaml`

### Version

main branch as of 2026-06-06

### Kernel version

_No response_

### OS / Distribution

_No response_

### Deployment

Bare metal / VM

--- a/.github/workflows/pull_request.yml
+++ b/.github/workflows/pull_request.yml
@@ -171,3 +171,55 @@ jobs:
           GOOS: ${{ matrix.goos }}
           GOARCH: ${{ matrix.goarch }}
         run: go build ./collector/...
+
+  otel-collector-example:
+    name: OTel Collector example (ocb build)
+    needs: generate-bpf
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    steps:
+      - uses: actions/checkout@df4cb1c069e1874edd31b4311f1884172cec0e10 # v6.0.3
+        with:
+          ref: ${{ inputs.ref || github.sha }}
+          persist-credentials: false
+
+      - name: Set up Go
+        uses: actions/setup-go@4a3601121dd01d1626a1e23e37211e3254c1c06c # v6.4.0 # zizmor: ignore[cache-poisoning] go.sum verifies module integrity
+        with:
+          go-version-file: "go.mod"
+          cache: true
+
+      - name: Go build cache
+        uses: actions/cache@27d5ce7f107fe9357f9df03efb73ab90386fccae # v5.0.5 # zizmor: ignore[cache-poisoning] go build cache is content-addressable and self-verifying
+        with:
+          path: ~/.cache/go-build
+          key: go-build-otelcol-example-${{ runner.os }}-${{ runner.arch }}-${{ hashFiles('**/*.go', 'go.sum') }}
+          restore-keys: |
+            go-build-otelcol-example-${{ runner.os }}-${{ runner.arch }}-
+
+      - name: Download generated BPF files
+        uses: actions/download-artifact@3e5f45b2cfb9172054b4087a40e8e0b5a5461e7c # v8.0.1
+        with:
+          name: bpf-generated-${{ github.run_id }}
+
+      - name: Resolve ocb version from go.mod
+        id: ocb
+        shell: bash
+        run: |
+          # Pin ocb to the Collector release line OBI itself depends on, so the
+          # example is always built against a compatible builder (see #2264).
+          version=$(grep -E '^[[:space:]]*go\.opentelemetry\.io/collector v[0-9]' go.mod | awk '{print $2}')
+          if [ -z "$version" ]; then
+            echo "could not determine collector version from go.mod" >&2
+            exit 1
+          fi
+          echo "version=$version" >> "$GITHUB_OUTPUT"
+
+      - name: Install ocb
+        env:
+          OCB_VERSION: ${{ steps.ocb.outputs.version }}
+        run: go install go.opentelemetry.io/collector/cmd/builder@"${OCB_VERSION}"
+
+      - name: Build collector distribution
+        working-directory: examples/otel-collector
+        run: "$(go env GOPATH)/bin/builder --config ./builder-config.yaml"

--- a/examples/otel-collector/README.md
+++ b/examples/otel-collector/README.md
@@ -5,7 +5,7 @@ This example demonstrates how to build and run the OpenTelemetry Collector with 
 ## Prerequisites
 
 - Go 1.25 or later
-- [OTel Collector Builder (`ocb`)](https://opentelemetry.io/docs/collector/extend/ocb/) **v0.151.0** installed. The component versions pinned in `builder-config.yaml` track the OpenTelemetry Collector `v0.151.0` release line (matching OBI's own `go.mod`). `ocb` is an unstable (`v0.x`) package with no cross-version compatibility guarantee, so use the `ocb` release that matches those component versions — a different `ocb` version may fail to build.
+- [OTel Collector Builder (`ocb`)](https://opentelemetry.io/docs/collector/extend/ocb/) installed. The `.github/workflows/pull_request.yml` workflow installs the version matching the OpenTelemetry Collector version in `go.mod`.
 - Docker (for generating eBPF files) or a C compiler, clang, and eBPF headers
 - Linux system with elevated privileges (sudo) to run the collector
 

--- a/examples/otel-collector/README.md
+++ b/examples/otel-collector/README.md
@@ -5,7 +5,7 @@ This example demonstrates how to build and run the OpenTelemetry Collector with 
 ## Prerequisites
 
 - Go 1.25 or later
-- [OTel Collector Builder (`ocb`)](https://opentelemetry.io/docs/collector/extend/ocb/) installed
+- [OTel Collector Builder (`ocb`)](https://opentelemetry.io/docs/collector/extend/ocb/) **v0.151.0** installed. The component versions pinned in `builder-config.yaml` track the OpenTelemetry Collector `v0.151.0` release line (matching OBI's own `go.mod`). `ocb` is an unstable (`v0.x`) package with no cross-version compatibility guarantee, so use the `ocb` release that matches those component versions — a different `ocb` version may fail to build.
 - Docker (for generating eBPF files) or a C compiler, clang, and eBPF headers
 - Linux system with elevated privileges (sudo) to run the collector
 

--- a/examples/otel-collector/builder-config.yaml
+++ b/examples/otel-collector/builder-config.yaml
@@ -8,22 +8,22 @@ replaces:
 
 exporters:
   - gomod:
-      go.opentelemetry.io/collector/exporter/debugexporter v0.145.0
+      go.opentelemetry.io/collector/exporter/debugexporter v0.151.0
   - gomod:
-      go.opentelemetry.io/collector/exporter/otlpexporter v0.145.0
+      go.opentelemetry.io/collector/exporter/otlpexporter v0.151.0
 
 processors:
   - gomod:
-      go.opentelemetry.io/collector/processor/batchprocessor v0.145.0
+      go.opentelemetry.io/collector/processor/batchprocessor v0.151.0
 
 receivers:
-  - gomod: go.opentelemetry.io/collector/receiver/otlpreceiver v0.145.0
+  - gomod: go.opentelemetry.io/collector/receiver/otlpreceiver v0.151.0
   - gomod: go.opentelemetry.io/obi v0.5.0
     import: go.opentelemetry.io/obi/collector
 
 providers:
-  - gomod: go.opentelemetry.io/collector/confmap/provider/envprovider v1.51.0
-  - gomod: go.opentelemetry.io/collector/confmap/provider/fileprovider v1.51.0
-  - gomod: go.opentelemetry.io/collector/confmap/provider/httpprovider v1.51.0
-  - gomod: go.opentelemetry.io/collector/confmap/provider/httpsprovider v1.51.0
-  - gomod: go.opentelemetry.io/collector/confmap/provider/yamlprovider v1.51.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/envprovider v1.57.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/fileprovider v1.57.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpprovider v1.57.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/httpsprovider v1.57.0
+  - gomod: go.opentelemetry.io/collector/confmap/provider/yamlprovider v1.57.0

